### PR TITLE
code corrections

### PR DESCRIPTION
**Pull request by guptasourabh**
Moved duplicate code at one place.
If condition needs to be in else if block.
Added errorMessaging in case of NPE exception

### CODE REVIEW

File: retrofit/src/main/java/retrofit2/OkHttpCall.java [hunk 1]
```diff
@@ -200,9 +200,7 @@ private void callSuccess(Response<T> response) {
       } finally {
         rawBody.close();
       }
-    }
-
-    if (code == 204 || code == 205) {
+    } else if (code == 204 || code == 205) {
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit2/OkHttpCall.java]: `else` is not needed after a `return`. Please revert.

File: retrofit-mock/src/main/java/retrofit2/mock/NetworkBehavior.java [hunk 2]
```diff
@@ -206,4 +200,10 @@ public long calculateDelay(TimeUnit unit) {
     long callDelayMs = (long) (delayMs * delayPercent);
     return MILLISECONDS.convert(callDelayMs, unit);
   }
+
+  private void checkPercentageValidity(int percentage, String message) {
```
Comment (JakeWharton) [inline retrofit-mock/src/main/java/retrofit2/mock/NetworkBehavior.java]: +`static`

**Comment by guptasourabh** [comment]
@JakeWharton changes done

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**